### PR TITLE
fix null value issue with gdal 2.2 (backport from 1.8)

### LINKED ISSUE #69: dependencies of 1.7.11 force multiple downgrades
Trying to upgrade to 1.7.11 currently forces a whole bunch of, I believe, unnecessary downgrades:

```bash
(stable) └─❱❱❱ conda update fiona                                                      +9025 18:15 ❰─┘
Solving environment: done

## Package Plan ##

  environment location: /Users/klay6683/miniconda3/envs/stable

  added / updated specs:
    - fiona


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    fiona-1.7.11               |           py36_1         622 KB  conda-forge
    cairo-1.14.6               |                4         1.2 MB  conda-forge
    fontconfig-2.12.1          |                4         266 KB  conda-forge
    glib-2.51.4                |                0         4.9 MB  conda-forge
    pillow-4.2.1               |           py36_0         504 KB  conda-forge
    gdal-2.1.4                 |           py36_2         643 KB  conda-forge
    matplotlib-2.1.0           |           py36_0         6.6 MB  conda-forge
    openjpeg-2.1.2             |                2         222 KB  conda-forge
    freetype-2.7               |                1         771 KB  conda-forge
    poppler-0.52.0             |                2         1.4 MB  conda-forge
    libgdal-2.1.4              |                9        12.7 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        29.9 MB

The following packages will be UPDATED:

    fiona:      1.7.10-py36h42caba7_0             --> 1.7.11-py36_1 conda-forge

The following packages will be DOWNGRADED:

    cairo:      1.14.10-0             conda-forge --> 1.14.6-4      conda-forge
    fontconfig: 2.12.6-0              conda-forge --> 2.12.1-4      conda-forge
    freetype:   2.8.1-0               conda-forge --> 2.7-1         conda-forge
    gdal:       2.2.3-py36_0          conda-forge --> 2.1.4-py36_2  conda-forge
    glib:       2.55.0-0              conda-forge --> 2.51.4-0      conda-forge
    libgdal:    2.2.3-2               conda-forge --> 2.1.4-9       conda-forge
    matplotlib: 2.1.2-py36_0          conda-forge --> 2.1.0-py36_0  conda-forge
    openjpeg:   2.3.0-2               conda-forge --> 2.1.2-2       conda-forge
    pillow:     5.0.0-py36_0          conda-forge --> 4.2.1-py36_0  conda-forge
    poppler:    0.61.1-3              conda-forge --> 0.52.0-2      conda-forge

Proceed ([y]/n)? n
```

System: macOS 10.12.6

### PR DESCRIPTION
**Pull request by akrherz**
fixes #69

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,7 +24,7 @@ requirements:
     - numpy 1.8.*  # [not (win and (py35 or py36))]
     - numpy 1.9.*  # [win and py35]
     - numpy 1.11.*  # [win and py36]
-    - gdal 2.1.*
+    - gdal 2.2.*
```
Comment (ocefpaf) [inline recipe/meta.yaml]: You'll need to update it below too.

**Comment by akrherz** [comment]
Augh, I think this will fail as I am patching a newer release :/  Sorry for the noise and will boggle some more

**Comment by ocefpaf** [comment]
> Augh, I think this will fail as I am patching a newer release :/ Sorry for the noise and will boggle some more

You'll probably need to refresh the patch. (Not an easy task!)

**Comment by akrherz** [comment]
> You'll probably need to refresh the patch. (Not an easy task!)

Eh, I gave it a shot, but a portion of the patch seemed to be conda-build 3 specific.

**Comment by ocefpaf** [comment]
> Eh, I gave it a shot, but a portion of the patch seemed to be conda-build 3 specific.

I'll take a look later this week to see what we can do.

**Comment by akrherz** [comment]
Eh, I was able to get everything green except appveyor :( Must be something to do with how the GDAL version is being set on windows and not triggering that extra code.  Since we are requiring gdal 2.2 already, lets just get rid of the check :)  I won't be insulted at all if you set all of my thrashing aside, I'm just desperate to get my conda-forge env moving forward again.

**Comment by akrherz** [comment]
Ha, I got them all green!  A broken clock is right twice a day :)

**Comment by ocefpaf** [comment]
Awesome! Thanks @akrherz! I'm merging this and, if everything goes as expected, I'll re-pin everything in conda-forge to use `gdal 2.2.*`.

**Pull request merged**